homeassistant/components/light/opple.py

```diff
@@ -0,0 +1,130 @@
+    vol.Optional(CONF_NAME, default="opple light"): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Opple light platform."""
+    name = config.get('name')
+    host = config.get('host')
+    add_devices([OppleLight(name, host)])
+
+
+class OppleLight(Light):
+    """Opple light device."""
+
+    def __init__(self, name, host):
+        """Initialize an Opple light."""
+        from pyoppleio.OppleLightDevice import OppleLightDevice
+        self._device = OppleLightDevice(host)
+
+        self._name = name
+        self._is_on = None
+        self._brightness = None
+        self._color_temp = None
+
+        _LOGGER.debug("init light %s %s", self._device.ip, self._device.mac)
+
+    @property
+    def available(self):
+        """Return True if light is available."""
+        return self._device.is_online
+
+    @property
+    def name(self):
+        """Return the display name of this light."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._is_on
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def color_temp(self):
+        """Return the color temperature of this light."""
+        return kelvin_to_mired(self._color_temp)
+
+    @property
+    def min_mireds(self):
+        """Return minimum supported color temperature."""
+        return 175
+
+    @property
+    def max_mireds(self):
+        """Return maximum supported color temperature."""
+        return 333
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        _LOGGER.debug("Turn on light %s %s", self._device.ip, kwargs)
+        if not self.is_on:
+            self._is_on = self._device.power_on = True
+
+        if ATTR_BRIGHTNESS in kwargs and \
+                self.brightness != kwargs[ATTR_BRIGHTNESS]:
+            self._brightness = kwargs[ATTR_BRIGHTNESS]
+            self._device.brightness = kwargs[ATTR_BRIGHTNESS]
+
+        if ATTR_COLOR_TEMP in kwargs and \
+                self.brightness != kwargs[ATTR_COLOR_TEMP]:
+            color_temp = mired_to_kelvin(kwargs[ATTR_COLOR_TEMP])
+            self._color_temp = self._device.color_temperature = color_temp
+
+    def turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        self._device.power_on = False
+        _LOGGER.debug("Turn off light %s", self._device.ip)
+
+    def update(self):
+        """Synchronize state with light."""
+        self._device.update()
+        self._is_on = self._device.power_on
+        self._brightness = self._device.brightness
+        self._color_temp = self._device.color_temperature
+
+        if not self.available:
```

Comment:
Only log one message when the device changes availability state, not on each update if state is the same.